providers/implementations/signature/rsa_sig.c

```diff
@@ -1541,3 +1932,259 @@ const OSSL_DISPATCH ossl_rsa_signature_functions[] = {
       (void (*)(void))rsa_settable_ctx_md_params },
     OSSL_DISPATCH_END
 };
+
+/* ------------------------------------------------------------------ */
+
+/*
+ * So called sigalgs (composite RSA+hash) implemented below.  They
+ * are pretty much hard coded, and rely on direct non-EVP use of hash
+ * implementations.
+ */
+
+static OSSL_FUNC_signature_query_key_types_fn rsa_sigalg_query_key_types;
+static OSSL_FUNC_signature_settable_ctx_params_fn rsa_sigalg_settable_ctx_params;
+static OSSL_FUNC_signature_set_ctx_params_fn rsa_sigalg_set_ctx_params;
+
+/*
+ * rsa_sigalg_signverify_init() is almost like rsa_digest_signverify_init(),
+ * just doesn't allow fetching an MD from whatever the user chooses.
+ */
+static int rsa_sigalg_signverify_init(void *vprsactx, void *vrsa,
+                                      OSSL_FUNC_signature_set_ctx_params_fn *set_ctx_params,
+                                      const OSSL_PARAM params[],
+                                      const OSSL_DISPATCH *mddispatch,
+                                      const char *mdname, int mdnid,
+                                      int operation, int pad_mode,
+                                      const char *desc)
+{
+    PROV_RSA_CTX *prsactx = (PROV_RSA_CTX *)vprsactx;
+
+    if (!ossl_prov_is_running())
+        return 0;
+
+    if (!rsa_signverify_init(vprsactx, vrsa, set_ctx_params, params, operation,
+                             desc))
+        return 0;
+
+    /* PSS is currently not supported as a sigalg */
+    if (prsactx->pad_mode == RSA_PKCS1_PSS_PADDING) {
+        ERR_raise(ERR_LIB_RSA, PROV_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE);
+        return 0;
+    }
+
+    if (!rsa_setup_md(prsactx, mdname, NULL, mddispatch, mdnid, desc))
+        return 0;
+
+    prsactx->pad_mode = pad_mode;
+    prsactx->flag_sigalg = 1;
+    prsactx->flag_allow_md = 0;
+
+    if (prsactx->mdctx == NULL) {
+        if (!wrap_md_newctx(prsactx))
+            goto error;
```

Comment:
Do we want to raise errors here?

Reply:
No.  It's assumed that the functions that wrap_md_newctx() calls already do.